test/test.jl

```diff
@@ -0,0 +1,25 @@
+using Llama2
+
+model = load_gguf_model("/llms/gguf/Meta-Llama-3-8B.Q4_K_S.gguf");
+
+#%%
+using CUDA
+using Llama2: to_cuda
+model_cu = nothing
+# we need to free cuda memory before loading a new model
+CUDA.reclaim(); GC.gc(true); #CUDA.memory_status()
+model_cu = to_cuda(model);
+
+#%%
+# GPU sample
+res = sample(model_cu, "Tim was happy."; temperature = 0.0f0, max_seq_len=10)
```

Comment:
For this test file, could we add like an `@test` call that matches one of the CPU-generated texts?

Reply:
Great idea! Going to add some test cases!

Reply:
After more carefully looking at this, I guess it is not that easy to test, as models with different versions give different results. The KA kernel solution for CPU and the same kernels on GPU are not giving the same token after ~40 tokens, and I guess the old CPU code gave not exactly the same results... but AFAI remember the first 10-20 tokens were pretty similar to what the current solution produces.